src/sonic-eventd/tools/events_tool.cpp

```diff
@@ -0,0 +1,362 @@
+      A value of 0 implies unlimited\n\
+\n\
+-p  - Count of milliseconds to pause between sends or receives. In send-recv mode, it only affects send.\n\
+      Default: 0 implying no pause\n\
+\n\
+      -i  - List of JSON messages to send in a file, with each event/message\n\
+            declared in a single line. When n is more than size of list, the list\n\
+            is rotated upon completion.\n\
+      e.g. '[ \n\
+                { \"sonic-bgp:bgp-state\": { \"ip\": \"10.101.01.10\", \"ts\": \"2022-10-11T01:02:30.45567\", \"state\": \"up\" }}\n\
+                { \"abc-xxx:yyy-zz\": { \"foo\": \"bar\", \"hello\":\"world\", \"ts\": \"2022-10-11T01:02:30.45567\"}}\n\
+                { \"some-mod:some-tag\": {}}\n\
+            ]\n\
+      Default: <some test message>\n\
+\n\
+-c  - Use offline cache in receive mode\n\
+-o  - O/p file to write received events\n\
+      Default: STDOUT\n";
+
+
+bool term_receive = false;
+
+template <typename Map>
+string
+t_map_to_str(const Map &m)
+{
+    stringstream _ss;
+    string sep;
+
+    _ss << "{";
+    for (const auto elem: m) {
+        _ss << sep << "{" << elem.first << "," << elem.second << "}";
+        if (sep.empty()) {
+            sep = ", ";
+        }   
+    }
+    _ss << "}";
+    return _ss.str();
+}
+
+void
+do_receive(const event_subscribe_sources_t filter, const string outfile, int cnt, int pause, bool use_cache)
+{
+    int index=0, total_missed = 0;
+    ostream* fp = &cout;
+    ofstream fout;
+
+    if (!outfile.empty()) {
+        fout.open(outfile);
+        if (!fout.fail()) {
+            fp = &fout;
+            printf("outfile=%s set\n", outfile.c_str());
+        }
+    }   
+    event_handle_t h = events_init_subscriber(use_cache, 2000, filter.empty() ? NULL : &filter);
+    printf("Subscribed with use_cache=%d timeout=2000 filter %s\n",
+            use_cache, filter.empty() ? "empty" : "non-empty");
+    ASSERT(h != NULL, "Failed to get subscriber handle");
+
+    while(!term_receive) {
+        event_receive_op_t evt;
+        map_str_str_t evtOp;
+
+        evt = event_receive(h);
+        if (evt.rc != 0) {
+            ASSERT(evt.rc == EAGAIN, "Failed to receive rc=%d index=%d\n",
+                    evt.rc, index);
+            continue;
+        }
+        ASSERT(!evt.key.empty(), "received EMPTY key");
+        ASSERT(evt.missed_cnt >= 0, "Missed count uninitialized");
+
+        total_missed += evt.missed_cnt;
+
+        evtOp[evt.key] = t_map_to_str(evt.params);
+        (*fp) << t_map_to_str(evtOp) << "\n";
+        fp->flush();
+        
+        if ((++index % PRINT_CHUNK_SZ) == 0) {
+            printf("Received index %d\n", index);
+        }
+
+        if (cnt > 0) {
+            if (--cnt <= 0) {
+                break;
+            }
+        }
+    }
+
+    events_deinit_subscriber(h);
+    printf("Total received = %d missed = %dfile:%s\n", index, total_missed,
+            outfile.empty() ? "STDOUT" : outfile.c_str());
+}
+
+
+#if 0
```

Comment:
why this is here? please provide description why this is disabled or remove

Reply:
Missed to remove. Now removed. Thank you!